drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java

```diff
@@ -63,27 +66,71 @@ public void tearDown() throws Exception {
-        assertEquals(1, conversionResult.getMessages().size());
-        assertTrue(conversionResult.getMessages().get(0).getMessage().startsWith("Created Guided Decision Table 'FunctionAndBigDecimalTable (converted on"));
-        assertTrue(conversionResult.getMessages().get(0).getMessage().endsWith(").gdst'"));
+        final String resourcePath = "dtables/src/main/resources/guvnor/feature/dtables/FunctionAndBigDecimalTable.xls";
+        final ConversionResult conversionResult = convertResource(resourcePath);
+        final List<ConversionMessage> messages = conversionResult.getMessages();
+        Assertions.assertThat(messages).hasSize(1);
+        final String message = messages.get(0).getMessage();
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(message).startsWith("Created Guided Decision Table 'FunctionAndBigDecimalTable (converted on");
+            softly.assertThat(message).endsWith(").gdst'");
+            softly.assertAll();
+        });
+    }
+
+    /**
+     * Covers RHBRMS-609 - Validation for spreadsheet fails because it cannot find functions in the same package
+     */
+    @Test
+    public void testValidateFunctionAndBigDecimal() throws Exception {
+        final String resourcePath = "dtables/src/main/resources/guvnor/feature/dtables/FunctionAndBigDecimalTable.xls";
+        final List<ValidationMessage> messages = validateResource(resourcePath);
+        Assertions.assertThat(messages).hasSize(0);
     }
 
     @Test
     public void testConvertStaticFields() throws Exception {
-        convertResource("dtables/src/main/resources/guvnor/feature/dtables/StaticFieldsTable.xls");
-        assertEquals(2, conversionResult.getMessages().size());
-        assertEquals("Unable to convert value 'Message.GOODBYE' to NUMERIC_INTEGER. Cell (F11)",
-                     conversionResult.getMessages().get(0).getMessage());
-        assertTrue(conversionResult.getMessages().get(1).getMessage().startsWith("Created Guided Decision Table 'StaticFieldsTable (converted on"));
-        assertTrue(conversionResult.getMessages().get(1).getMessage().endsWith(").gdst'"));
+        final String resourcePath = "dtables/src/main/resources/guvnor/feature/dtables/StaticFieldsTable.xls";
+        final ConversionResult conversionResult = convertResource(resourcePath);
+        final List<ConversionMessage> messages = conversionResult.getMessages();
+        Assertions.assertThat(messages).hasSize(2);
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(messages.get(0).getMessage())
+                    .isEqualTo("Unable to convert value 'Message.GOODBYE' to NUMERIC_INTEGER. Cell (F11)");
+            softly.assertThat(messages.get(1).getMessage())
+                    .startsWith("Created Guided Decision Table 'StaticFieldsTable (converted on");
+            softly.assertThat(messages.get(1).getMessage())
+                    .endsWith(").gdst'");
+            softly.assertAll();
```

Comment:
This last line is not necessary. Happens automatically at the end of the `assertSoftly()` call.